docs/cpu-constraints.md

```diff
@@ -0,0 +1,40 @@
+# CPU Constraints in Clear Containers
+
+## Default number of virtual CPUs
+
+Before starting a container, the runtime reads the option `default_vcpus` from the configuration file to determine the number of virtual CPUs (vCPUs) needed to start the virtual machine. By default, `default_vcpus` is equal to 1 for fast boot time and a small memory footprint per virtual machine. Be aware that increasing this value negatively impacts the virtual machine's boot time and memory footprint.
+In general, we recommend that you do not edit this variable, unless you know what are you doing. If your container needs more than one vCPU, use [docker `--cpus`][1], [docker update][4] or [kubernetes `cpu` limits][2] to assign more resources.
+
+## Virtual CPUs and Kubernetes PODs
+
+A kubernetes POD is a group of one or more containers, with shared storage and network, and a specification for how to run the containers [[1][3]]. In Clear Containers this group of containers run inside the same virtual machine. If you do not specify a CPU constraint, the runtime does not hot add more vCPUs and the container is not placed inside a CPU cgroup.  Instead, the container uses the number of vCPUs specified by `default_vcpus` and shares these resources with other containers in the same situation (without a CPU constraint).
+
+## Containers life cycle
+
+When you create a container with a CPU constraint, the runtime hot adds the number of vCPUs required by the container. Similarly, when the container stops, the runtime hot removes these resources.
+
+## Containers without CPU constraint
+
+A container without a CPU constraint uses the default number of vCPUs specified in the configuration file, in the case of Kubernetes PODs, containers without a CPU constraint use and share between them the default number of vCPUs, for example, if `default_vcpus` is equal to 1 and you have 2 containers without CPU constraint and each container tries to consume 100% of vCPU, since your virtual machine has not enough resources to satisfy containers needs, the resources are divided in two parts, 50% of vCPU for each container. If you wish to give access to a greater or lesser portion of vCPUs to a specific container, you can use [docker --cpu-shares][1] or [kubernetes `cpu` requests][2].
```

Comment:
Suggested rewrite to break up this sentence: "A container without a CPU constraint uses the default number of vCPUs specified in the configuration file. In the case of Kubernetes PODs, containers without a CPU constraint use and share between them the default number vCPUs. For example, if `default_vcpus` is equal to 1 and you have 2 containers without CPU constraints with each container trying to consume 100% of vCPU, the resources divide in two parts, 50% of vCPU for each container because your virtual machine does not have enough resources to satisfy containers needs."